common/src/jni/main/cpp/conscrypt/jniutil.cc

```diff
@@ -352,12 +352,14 @@ int throwForEvpError(JNIEnv* env, int reason, const char* message,
         case EVP_R_MISSING_PARAMETERS:
         case EVP_R_INVALID_PEER_KEY:
         case EVP_R_DECODE_ERROR:
+        case EVP_R_NOT_A_PRIVATE_KEY:
```

Comment:
FWIW, you will never get this error in the function you've added. This error exists because OpenSSL uses a single type for public and private keys, and so we have to rely on a runtime check. But `EVP_parse_private_key` will never give you a public key.

Reply:
Good point.  I did contrive a scenario for this, because the current XDH usage I have seen is apps storing raw XEC keys, so it's not entirely infeasible that a bug or user error causes them to accidentally wrap public key with a PKCS#8 preamble and later try to decode it.

Tracing the x25519 part of the code, it looks like that would get at least as far as `x25519_set_priv_raw` and `X25519_public_from_private` (which would presumably generate rubbish) but then I guess the EVP code would error out?